contributed_definitions/nyaml/NXpeak.yaml

```diff
@@ -1,59 +1,140 @@
   position(NX_NUMBER):
     unit: NX_ANY
     doc: |
       In the case of an empirical description of the peak and its shoulders,
       this array holds the position values for the independent variable.
-    dimensions:
-      rank: 1
-      dim: [[1, n_support]]
+    dim: (n_support,)
   intensity(NX_NUMBER):
     unit: NX_ANY
     doc: |
       In the case of an empirical description of the peak and its shoulders,
       this array holds the intensity/count values at each position.
-    dimensions:
-      rank: 1
-      dim: [[1, n_support]]
-  (NXcollection):
+    dim: (n_support,)
+  width(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      Width of a peak at a defined fraction of the peak height.
+      
+      Usually, this will be the Full Width at Half Maximum of the peak (FWHM).
+      For asymmetric peaks, convenient measures of peak width are the half-widths of
+      each side of the peak at half maximum intensity.
+    \@type:
+      doc: |
+        Specification of the definition of `width`.
+      enumeration:
+        fwhm:
+          doc: |
+            Full Width at Half Maximum of the peak.
+        hwhm:
+          doc: |
+            Half Width at Half Maximum of the peak.
+        half_widths_asymmetric:
+          doc: |
+            For asymmetrical peaks, the half-widths of each side of the peak at
+            half maximum intensity is given.
+        other:
+          doc: |
+            For other peak width definitions.
+  functional_form:
     doc: |
-      In the case of an analytical description (or if peak_model is other) this
+      A formula to describe the line shape using set of parameters as entered
+      by the `peak_parameters` fields.
+  peak_parameters(NXparameters):
+    doc: |
+      In the case of an analytical description (or if description is other) this
       collection holds parameter of (and eventually) the functional form.
       For example in the case of Gaussians mu, sigma, cut-off values,
       and background intensity are relevant parameter.
+  total_area(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      Total area under the curve (can also be used for the total area minus any
+      background values.)
+  relative_area(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      Relative area of this peak compared to other peaks in an NXpeak_model.
+      
+      The relative area can simply be derived by dividing the total_area by the total
+      area of all peaks or by a more complicated method (e.g., by additionally dividing
+      by the relative_sensitivity_factor). Details should be stored in the quantification
+      description of NXpeak_model.
+  relative_sensitivity_factor(NX_NUMBER):
+    unit: NX_UNITLESS
+    doc: |
+      Relative sensitivity for this peak, to be used in quantification in an NXpeak_model.
+      
+      As an example, in X-ray spectroscopy could depend on the energy scale (see position),
+      the ionization cross section, and the element probed.
```

Comment:
These creates an interdependence to the NXpeak_model, not sure I find this a good idea to place it here.
Can you not add these fields inside the NXpeak instances inside NXpeak_model?